src/cookbook/games/firestore-multiplayer.md

```diff
@@ -0,0 +1,469 @@
+the `card` template that you'll find
+in the [`flutter/games` repository](https://github.com/flutter/games).
+Run the following command to clone that repository:
+
+```shell
+$ git clone https://github.com/flutter/games.git
+```
+
+{% comment %}
+  If/when we have a "sample_extractor" tool, or any other easier way
+  to get the code, mention that here.
+{% endcomment %}
+
+Open the project in `templates/card`.
+
+{{ site.alert.note }}
+  You can ignore this step and follow the recipe with your own game
+  project. Adapt the code at appropriate places.
+{{ site.alert.end }}
+
+
+## 2. Install Firestore
+
+[Cloud Firestore](https://cloud.google.com/firestore/)
+is a horizontally scaling NoSQL document database in the cloud.
+It includes built-in live synchronization.
+This is perfect for our needs.
+It keeps the game state updated in the cloud database,
+so every player sees the same state.
+
+If you want a quick, 15-minute primer on Cloud Firestore,
+watch the following video:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/v_hR4K4auoQ" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
+
+To add Firestore to your Flutter project,
+follow the first two steps of the
+[Get started with Cloud Firestore](https://firebase.google.com/docs/firestore/quickstart)
+guide.
+The desired outcomes include:
+
+-   A Firestore database ready in the cloud, in **Test mode**
+-   A generated `firebase_options.dart` file
+-   The appropriate plugins added to your `pubspec.yaml`
+
+You *don't* need to write any Dart code in this step.
+As soon as you get the step of writing Dart code in that guide,
+return to this recipe.
+
+{% comment %}
+  Revisit to see if we can inline the steps here:
+  https://firebase.google.com/docs/flutter/setup
+  ... followed by the first 2 steps here:
+  https://firebase.google.com/docs/firestore/quickstart
+{% endcomment %}
+
+
+## 3. Initialize Firestore
+
+1.  Open `lib/main.dart` and import the plugins,
+    as well as the `firebase_options.dart` file
+    that was generated by `flutterfire configure` in the previous step.
+
+    ```dart
+    import 'package:cloud_firestore/cloud_firestore.dart';
+    import 'package:firebase_core/firebase_core.dart';
+    
+    import 'firebase_options.dart';
+    ```
+
+2.  Add the following code just above the call to `runApp()`
+    in `lib/main.dart`:
+
+    ```dart
+     WidgetsFlutterBinding.ensureInitialized();
+    
+      await Firebase.initializeApp(
+        options: DefaultFirebaseOptions.currentPlatform,
+      );
+    ```
+
+    This ensures that Firebase is initialized on game startup.
+
+3.  Add the Firestore instance to the app.
+    That way, any widget can access this instance.
+    Widgets can also react to the instance missing if needed.
+
+    Replace the usual `runApp(MyApp())` with the following:
```

Comment:
I feel this step jumps a bit. It's mentioned earlier these steps can be followed without the card template, but this step assumes the user has provider as a dependency and they know what it is/what it's doing.

Could this be expanded on a bit or at least the requirement mentioned (`dart pub add provider`)?

Reply:
Good point, I'll address this.